src/client/App.js

```diff
@@ -5,28 +5,39 @@ import Meals from "./components/Meals";
 import Reservations from "./components/Reservations";
 import Reviews from "./components/Reviews";
 import About from "./components/About";
+import AddMeal from "./components/AddMeal";
+import AddReview from "./components/AddReview";
+import MealById from "./components/MealById";
+import ReviewById from "./components/ReviewById";
+import AutoSearch from "./components/AutoSearch";
+import AllMeals from "./components/All-meals";
 import { ContextProvider } from "./components/DataContext";
 import "./App.css";
 
 function App() {
   return (
     <Router className="app">
       <Switch>
-        <Route exact path="/home" component={HomePage} />
+        <Route exact path="/" component={HomePage} />
         <Route exact path="/about" component={About} />
         <Route exact path="/meals" component={Meals} />
-        <Route
-          exact
-          path="/reservations"
-          component={Reservations}
-        />
+        <Route exact path="/all-meals" component={AllMeals} />
+        <Route exact path="/reservations" component={Reservations} />
         <Route exact path="/reviews" component={Reviews} />
+        <Route exact path="/meals/addMeal" component={AddMeal} />
+        <Route exact path="/reviews/addReview" component={AddReview} />
+        <Route exact path="/meals/:id" component={MealById} />
+        <Route exact path="/reviews/:id" component={ReviewById} />
+        <Route exact path="/searchMeal" component={AutoSearch} />
         <ContextProvider>
           <Meals />
+          <AllMeals />
+          <Reviews />
+          <MealById />
+          <Reservations />
         </ContextProvider>
```

Comment:
I'm confused why this is here.

These are all components that seem to be pages, why are they not defined as a <Route>?
Is it because you needed access to the data context? 
You most likely want to wrap the data context around the <Switch> instead if you want to to be available for all children of your app.

Like this, these components won't be rendered the right way.